utils/fixed_decimal/src/decimal.rs

```diff
@@ -350,65 +351,103 @@ impl FixedDecimal {
     /// let mut dec = FixedDecimal::from(42);
     /// assert_eq!("42", dec.to_string());
     ///
-    /// dec.pad_or_truncate_left(2);
+    /// dec.padded_left(3);
     /// assert_eq!("0042", dec.to_string());
     ///
-    /// dec.pad_or_truncate_left(-2);
-    /// assert_eq!("42", dec.to_string());    
+    /// dec.padded_left(2);
+    /// assert_eq!("042", dec.to_string());
     ///
-    /// dec.pad_or_truncate_left(-1);
-    /// assert_eq!("2", dec.to_string());
+    /// dec.padded_left(1);
+    /// assert_eq!("42", dec.to_string());
+    ///
+    /// dec.padded_left(0);
+    /// assert_eq!("42", dec.to_string());
     /// ```
-    pub fn pad_or_truncate_left(&mut self, shift: i16) {
-        self.upper_magnitude = match shift >= 0 {
-            true => self.upper_magnitude.checked_add(shift).unwrap_or(i16::MAX),
-            false => self.upper_magnitude.checked_add(shift).unwrap_or(0),
-        };
-
-        // upper_magnitude must be greater than or equal zero.
-        if self.upper_magnitude < 0 {
-            self.upper_magnitude = 0;
-        }
-
-        if self.upper_magnitude >= self.magnitude {
-            return;
+    pub fn padded_left(&mut self, magnitude: u16) {
```

Comment:
Nit: If the argument is "number of integer digits", then `u16` makes some sense.  But if the argument is "power of 10", then I prefer to use `i16`, with all negative values treated as 0.